src/build_helper.rs

```diff
@@ -0,0 +1,262 @@
+use core::ops::Range;
+
+use alloc::vec::Vec;
+
+use crate::errors::{DaachorseError, Result};
+
+const INVALID_IDX: u32 = u32::MAX;
```

Comment:
Could you consider using Some(u32) instead of this constant?

Reply:
https://github.com/daac-tools/daachorse/pull/48/commits/d1489beac4dda3cfbcf7d4619a4ab81bfb89fda6